crates/unftp-sbe-fs/src/lib.rs

```diff
@@ -69,48 +81,35 @@ impl Filesystem {
     /// asks for `hello.txt`, the server will send it `/srv/ftp/hello.txt`.
     pub fn new<P: Into<PathBuf>>(root: P) -> Self {
         let path = root.into();
-        Filesystem {
-            root: canonicalize(&path).unwrap_or(path),
-        }
-    }
-
-    /// Returns the full, absolute and canonical path corresponding to the (relative to FTP root)
-    /// input path, resolving symlinks and sequences like '../'.
-    async fn full_path<P: AsRef<Path>>(&self, path: P) -> Result<PathBuf> {
-        // `path.join(other_path)` replaces `path` with `other_path` if `other_path` is absolute,
-        // so we have to check for it.
-        let path = path.as_ref();
-        let full_path = if path.starts_with("/") {
-            self.root.join(path.strip_prefix("/").unwrap())
-        } else {
-            self.root.join(path)
-        };
-
-        let real_full_path = tokio::task::spawn_blocking(move || canonicalize(full_path))
-            .await
-            .map_err(|e| Error::new(ErrorKind::LocalError, e))??;
-
-        if real_full_path.starts_with(&self.root) {
-            Ok(real_full_path)
-        } else {
-            Err(Error::from(ErrorKind::PermanentFileNotAvailable))
-        }
+        let aa = cap_std::ambient_authority();
+        let root_fd = Arc::new(cap_std::fs::Dir::open_ambient_dir(&path, aa).unwrap());
+        Filesystem { root_fd, root: path }
     }
 }
 
 #[async_trait]
 impl<User: UserDetail> StorageBackend<User> for Filesystem {
     type Metadata = Meta;
 
+    fn enter<U: UserDetail>(&mut self, user_detail: &U) -> io::Result<()> {
```

Comment:
No need to define `U: UserDetail` here, we can just use `User` that is defined on line 91. I can fix it post merge.